apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet/Trace/SimpleManagedTracer.cs

```diff
@@ -145,6 +145,23 @@ public string GetCurrentTraceId()
             return _trace.TraceId;
         }
 
+        /// <inheritdoc />
+        public ulong? GetCurrentSpanId()
+        {
+            lock (_stackMutex)
```

Comment:
My only concern is that a request can possibly have the locking overhead 4 or more times: for StartSpan, EndSpan, AnnotateSpan, and GetCurrentSpanId. But perhaps it is best to not optimize prematurely here.